### PR TITLE
RTL: Add approaches for rtl to home and rally points

### PR DESCRIPTION
**Pull request by KonradRudin**
### Solved Problem
When a VTOL performs a RTL to the home or rally position, it flies straight to the designated position and lands there. If it is too high, it will circle around the designated position. however, it is desired to actually define a landing pattern with a designated loiter position when performing a RTL for a vtol in Fixed wing mode.

Fixes #{Github issue ID}

### Solution
- Using the MAVLINK Safe point messages, we can not only upload safe point landing locations, but designated loiter positions as well. Multiple loiter points can be defined for the safe point. The storage order of the loiter points determine to which safe landing position the loiter point belongs to.
- Update the RTL logic to check if there exists some loiter position for the rtl landing point. If it exists, choose the best approach best on current vehicle location and wind condition and the perform a mission landing approach.

### Test coverage
- Unit/integration test: tested in SITL with standard_vtol
- Simulation/hardware testing logs: 
  - https://logs.px4.io/plot_app?log=7db47a76-72a2-417e-a9ac-70ff1f6f1794
  - https://logs.px4.io/plot_app?log=fd874a72-2c96-40e1-bd49-71695687dbd8

### Context
Related links, screenshot before/after, video

### CODE REVIEW

File: src/modules/navigator/rtl.cpp [hunk 1]
```diff
@@ -185,31 +209,58 @@ void RTL::setRtlTypeAndDestination()
...
-		} else {
+		case DestinationType::DESTINATION_TYPE_SAFE_POINT:
 			_rtl_direct.setRtlAlt(rtl_alt);
 			_rtl_direct.setRtlPosition(rtl_position);
 			_rtl_type = RtlType::RTL_DIRECT;
+			break;
+
+		case DestinationType::DESTINATION_TYPE_HOME: {
+				// check if we can apply vtol land.
+				bool vtol_land{_vehicle_status_sub.get().is_vtol &&
```
Comment (RomanBapst) [inline src/modules/navigator/rtl.cpp:231]:
```suggestion
				const bool vtol_land{_vehicle_status_sub.get().is_vtol &&
```

File: src/modules/navigator/vtol_land.cpp [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+	return _vtol_home_land_approaches.isAnyApproachValid();
+}
+
+void
+VtolLand::set_loiter_position()
+{
+
+	_land_approach = chooseBestLandingApproach();
+
+	if (PX4_ISFINITE(_land_approach.lat) && PX4_ISFINITE(_land_approach.lon)) {
```
Comment (RomanBapst) [inline src/modules/navigator/vtol_land.cpp:194]:
```suggestion
	if (_land_approach.isValid()) {
```

File: src/modules/navigator/rtl.h [hunk 3]
```diff
@@ -65,8 +66,10 @@ class RTL : public NavigatorMode, protected PlannedMissionInterface, public Modu
 
 	enum class RtlType {
 		RTL_DIRECT,
+		RTL_DIRECT_SAFE_VTOL,
 		RTL_MISSION_FAST,
 		RTL_MISSION_FAST_REVERSE,
+		VTOL_LAND
```
Comment (RomanBapst) [inline src/modules/navigator/rtl.h:72]: @KonradRudin What is this guy used for?

File: src/modules/navigator/vtol_land.cpp (hunk 2 above)
Comment (RomanBapst) [reply src/modules/navigator/vtol_land.cpp:194]: Hm, although the altitude is kind of separate.

File: src/modules/navigator/vtol_land.cpp [hunk 4]
```diff
@@ -0,0 +1,329 @@
...
+			_vtol_home_land_approaches.approaches[sector_counter].height_m = mission_item.altitude;
+			_vtol_home_land_approaches.approaches[sector_counter].loiter_radius_m = mission_item.loiter_radius;
+			sector_counter++;
+		}
+	}
+}
+
+rtl_time_estimate_s VtolLand::calc_rtl_time_estimate()
+{
+	rtl_time_estimate_s time_estimate;
```
Comment (RomanBapst) [inline src/modules/navigator/vtol_land.cpp:324]: @KonradRudin This is awesome, others have used this complicated logic using wind and distance and altitude but your algorithm is much simpler, this is the way to go!

**Comment by KonradRudin** [comment]
I put https://github.com/PX4/PX4-Autopilot/pull/21049/commits/2a5c60ca70c3c4d96480f60acf47246d8b894c35 and https://github.com/PX4/PX4-Autopilot/pull/21049/commits/b898bf37b8b834a25af1abcc4e713719a5634b9c in a separate PR [here](https://github.com/PX4/PX4-Autopilot/pull/22230) since those are fixes not related to the rtl home approaches

**Comment by KonradRudin** [comment]
rebased on main and squashed commits

File: src/modules/navigator/rtl.h (hunk 3 above)
Comment (KonradRudin) [reply src/modules/navigator/rtl.h]: Not used anymore

File: src/modules/navigator/vtol_land.cpp (hunk 4 above)
Comment (KonradRudin) [reply src/modules/navigator/vtol_land.cpp]: Fixed

**Comment by KonradRudin** [comment]
Added flight logs in the description

**Review by RomanBapst — APPROVED** [review]

**Pull request merged**